classes/webservice/ApiNode.php

```diff
@@ -0,0 +1,378 @@
+    }
+
+    /**
+     * @param array $attributes
+     *
+     * @return self
+     */
+    public function setAttributes(array $attributes): self
+    {
+        $this->attributes = $attributes;
+
+        return $this;
+    }
+
+    /**
+     * @param array $nodes
+     *
+     * @return self
+     */
+    public function setNodes(array $nodes): self
+    {
+        $this->nodes = $nodes;
+
+        return $this;
+    }
+
+    /**
+     * @param string $name
+     * @param string|null $value
+     *
+     * @return self
+     */
+    public function addAttribute(string $name, ?string $value = null): self
+    {
+        $this->attributes[$name] = $value;
+
+        return $this;
+    }
+
+    /**
+     * Appends $node as child to current ApiNode
+     *
+     * @param ApiNode $node
+     *
+     * @return ApiNode self
+     */
+    public function addApiNode(ApiNode $node): self
+    {
+        $this->nodes[] = $node;
+
+        return $this;
+    }
+
+    /**
+     * Create new ApiNode of type "value" and appends it as child to current ApiNode
+     *
+     * @param string $name
+     * @param string|null $value
+     *
+     * @return ApiNode Created child node
+     */
+    public function addNode(string $name, ?string $value = null): self
+    {
+        $newNode = self::value($name, $value);
+        $this->nodes[] = $newNode;
+
+        return $newNode;
+    }
+
+    /**
+     * Create new ApiNode of type "lang" and appends it as child to current ApiNode
+     *
+     * @param string $name
+     *
+     * @return ApiNode Created child node
+     */
+    public function addLanguageNode(string $name): self
+    {
+        $newNode = self::lang($name);
+        $this->nodes[] = $newNode;
+
+        return $newNode;
+    }
+
+    /**
+     * Create new ApiNode of type "parent" and appends it as child to current ApiNode
+     *
+     * @param string $name
+     * @param array $attributes
+     *
+     * @return ApiNode Created child node
+     */
+    public function addParentNode(string $name = null, array $attributes = []): self
+    {
+        $newNode = self::parent($name, $attributes);
+        $this->nodes[] = $newNode;
+
+        return $newNode;
+    }
+
+    /**
+     * Create new ApiNode of type "list" and appends it as child to current ApiNode
+     *
+     * @param string $name
+     * @param array $attributes
+     *
+     * @return ApiNode Created child node
+     */
+    public function addListNode(string $name = null, array $attributes = [])
+    {
+        $newNode = self::list($name, $attributes);
+        $this->nodes[] = $newNode;
+
+        return $newNode;
+    }
+
+    /**
+     * Transform $field array into ApiNode and appends it as child to current node
+     *
+     * @param array $field
+     * @param string|null $schemaToDisplay
+     *
+     * @return self
+     */
+    public function addField(array $field, ?string $schemaToDisplay = null): self
+    {
+        $newNode = self::value($field['sqlId']);
```

Comment:
Can you describe in the phpDoc the keys that can exist in the $field array ?

Reply:
Most of this logic is not changed and taken from previous api, which has not been documented at all :neutral_face: . I tried to describe the array inputs from what I found in code directly, please check 885dba5 whether its correct